Modules/Forum/classes/class.ilObjForumAdministrationGUI.php

```diff
@@ -92,143 +100,113 @@ public function getAdminTabs(): void
+            'send_attachments_by_mail',
+            'enable_fora_statistics',
+            'enable_anonymous_fora',
+            'save_post_drafts',
+            'forum_default_view',
+            'forum_notification',
+        ]);
+
+        $drafts = null !== $data[$drafts_key];
+        $this->settings->set('autosave_drafts', (string) ((int) $drafts));
+        if ($drafts) {
+            $this->settings->set('autosave_drafts_ival', (string) ((int) $data['autosave_drafts']['ival']));
+        }
 
         $this->tpl->setOnScreenMessage('success', $this->lng->txt('settings_saved'));
-        $form->setValuesByPost();
         $this->editSettings($form);
     }
 
-    protected function populateForm(ilPropertyFormGUI $form): void
+    protected function settingsForm(): Form
     {
-        $frma_set = new ilSetting('frma');
-
-        $form->setValuesByArray([
-            'fora_statistics' => (bool) $this->settings->get('enable_fora_statistics'),
-            'anonymous_fora' => (bool) $this->settings->get('enable_anonymous_fora'),
-            'forum_notification' => (int) $this->settings->get('forum_notification', '0') === 1,
-            'file_upload_allowed_fora' => (int) $this->settings->get(
-                'file_upload_allowed_fora',
-                (string) ilForumProperties::FILE_UPLOAD_GLOBALLY_ALLOWED
-            ),
-            'save_post_drafts' => (int) $this->settings->get('save_post_drafts', '0'),
-            'autosave_drafts' => (int) $this->settings->get('autosave_drafts', '0'),
-            'autosave_drafts_ival' => (int) $this->settings->get('autosave_drafts_ival', '30'),
-            'send_attachments_by_mail' => (bool) $this->settings->get('send_attachments_by_mail')
-        ]);
-    }
+        $field = $this->ui->factory()->input()->field();
 
-    protected function getSettingsForm(): ilPropertyFormGUI
-    {
-        $form = new ilPropertyFormGUI();
-        $form->setFormAction($this->ctrl->getFormAction($this, 'saveSettings'));
-        $form->setTitle($this->lng->txt('settings'));
-
-        $check = new ilCheckboxInputGUI($this->lng->txt('enable_fora_statistics'), 'fora_statistics');
-        $check->setInfo($this->lng->txt('enable_fora_statistics_desc'));
-        $form->addItem($check);
-
-        $check = new ilCheckboxInputGUI($this->lng->txt('enable_anonymous_fora'), 'anonymous_fora');
-        $check->setInfo($this->lng->txt('enable_anonymous_fora_desc'));
-        $form->addItem($check);
+        $items = [];
 
-        $file_upload = new ilRadioGroupInputGUI(
-            $this->lng->txt('file_upload_allowed_fora'),
-            'file_upload_allowed_fora'
+        $checkbox = fn ($label) => $field->checkbox($this->lng->txt($label), $this->lng->txt($label . '_desc'));
+        $nice = fn ($label) => sprintf('%s (%s)', $this->lng->txt('sort_by_date'), $this->lng->txt($label));
+        $to_string = static fn ($x) => (string) $x;
+        $withOptions = static fn ($x, $options) => array_reduce(
```

Comment:
New variable names should be `snake_case`.

Reply:
Done